src/state_machine.zig

```diff
@@ -533,8 +533,41 @@ pub fn StateMachineType(
             callback(self);
         }
 
+        pub fn input_valid(operation: Operation, input: []align(16) const u8) bool {
```

Comment:
Could we implement this validation without increasing the surface area of the state machine? e.g.:

- Add `get_account_transfers`, `get_account_balances` to `batch_max` (hard coded to 1 each).
- Replica uses `constants.batch_max` and `@sizeOf(Event)` to determine valid sizes.

Reply:
I went back and forth deciding whether the validation should be done by the replica or the state machine.
My reasons why I implemented at the state machine level:
- We would need to split `batch_max` into `event_max` and `result_max` (which I think is a good change).
- The state machine may need to perform something more complex than simply validating the length.
- We can add `assert(input_valid(...))` everywhere in the state machine code :smile:.

Reply:
I thought I posted a response to this already but I can't find it, so apologies if this is duplicated...

> The state machine may need to perform something more complex than simply validating the length.

Good point. These are "hard" errors, though -- they terminate the session and the operator must restart the client.
Maybe that's ok though... in the future, `input_valid()` could return an error message, and we could include that error message in the `Eviction` body to bubble it back up to the client/operator.

(Currently error reporting is not very transparent -- the two error behaviors we have are "client retries endlessly but server keeps ignoring it" and "the server sends an eviction to the client and the client panics".)